include/s3select.h

```diff
@@ -2132,49 +2138,86 @@ class base_s3object
 	return m_returned_bytes_size;
   }
 
-  void result_values_to_string(multi_values& projections_resuls, std::string& result)
+  void json_result_format(multi_values& projections_results, std::string& result, std::string& output_delimiter, std::string& last_key)
   {
-    size_t i = 0;
+    result += "{";
+    for (size_t i = 0; i < projections_results.values.size(); ++i)
+    {
+      auto& res = projections_results.values[i];
+      std::string label = "_";
+      label += std::to_string(i + 1);
+
+      if (i > 0) {
+        result += output_delimiter;
+      }
+      if(!is_star && m_aggr_flow)  {
+        result += "\"" + label + "\":";
+        } else if(!is_star && !m_aggr_flow) {
+          result += "\"" + last_key + "\":";
+      }
+
+      result.append(res->to_string());
+      m_returned_bytes_size += strlen(res->to_string());
+      }
+    result += "}";
+  }
+
+
+  void result_values_to_string(multi_values& projections_resuls, std::string& result)
+{   
+    std::string last_key;
     std::string output_delimiter(1,m_csv_defintion.output_column_delimiter);
     std::string output_row_delimiter(1,m_csv_defintion.output_row_delimiter);
+    std::vector<std::string> key_path(m_sa->get_json_key_path());
```

Comment:
per each call to `result_values_to_string` the `key_path` is set.
it should be avoided, this function is quit intensive.
the same with `last_key`

all of this should reside on `json_result_format`